client/verta/tests/unit_tests/pipeline/test_pipeline_graph.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+"""
+Unit tests for the PipelineGraph class
+"""
+
+from unittest.mock import patch
+
+from hypothesis import given, HealthCheck, settings, strategies as st
+
+import verta
+from tests.unit_tests.strategies import pipeline_definition
+from verta.pipeline import PipelineGraph
+
+
+def test_set_steps(make_mock_pipeline_step, make_mock_registered_model) -> None:
+    """Test that the steps of a PipelineGraph can be set."""
+    mocked_rm = make_mock_registered_model(id=123, name="test_rmv")
+    with patch.object(
+        verta.pipeline.PipelineStep, "_get_registered_model", return_value=mocked_rm
+    ):
+        step_1 = make_mock_pipeline_step()
+        step_2 = make_mock_pipeline_step()
+        graph = PipelineGraph(steps=set())
+    graph.set_steps({step_1, step_2})
+    assert set(graph.steps) == {step_1, step_2}
+    graph.set_steps(set())
+    assert not graph.steps
+
+
+@given(
+    pipeline_definition=pipeline_definition(),
+    registered_model_id=st.integers(min_value=1, max_value=2**63),
+    # max value limit avoids protobuf "Value out of range" error
```

Comment:
Recommend swapping the lines so that the comment is above the code in question

```suggestion
    # max value limit avoids protobuf "Value out of range" error
    registered_model_id=st.integers(min_value=1, max_value=2**63),
```